app/src/main/java/org/oppia/android/app/devoptions/marktopicscompleted/testing/MarkTopicsCompletedTestActivity.kt

```diff
@@ -0,0 +1,46 @@
+package org.oppia.android.app.devoptions.marktopicscompleted.testing
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import org.oppia.android.R
+import org.oppia.android.app.activity.InjectableAppCompatActivity
+import org.oppia.android.app.devoptions.marktopicscompleted.MarkTopicsCompletedFragment
+
+/** The activity for testing [MarkTopicsCompletedFragment] */
+class MarkTopicsCompletedTestActivity : InjectableAppCompatActivity() {
+
+  private var internalProfileId = -1
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    activityComponent.inject(this)
+    supportActionBar?.setDisplayHomeAsUpEnabled(true)
+    supportActionBar?.setHomeAsUpIndicator(R.drawable.ic_arrow_back_white_24dp)
+    setContentView(R.layout.mark_topics_completed_activity)
+    internalProfileId = intent.getIntExtra(MARK_TOPICS_COMPLETED_TEST_ACTIVITY_PROFILE_ID_KEY, -1)
+    if (getMarkTopicsCompletedFragment() == null) {
+      val markTopicsCompletedFragment = MarkTopicsCompletedFragment.newInstance(internalProfileId)
+      supportFragmentManager.beginTransaction().add(
+        R.id.mark_topics_completed_container,
+        markTopicsCompletedFragment
+      ).commitNow()
+    }
+  }
+
+  private fun getMarkTopicsCompletedFragment(): MarkTopicsCompletedFragment? {
+    return supportFragmentManager
+      .findFragmentById(R.id.mark_topics_completed_container) as MarkTopicsCompletedFragment?
+  }
+
+  companion object {
+    const val MARK_TOPICS_COMPLETED_TEST_ACTIVITY_PROFILE_ID_KEY =
+      "MarkTopicsCompletedTestActivity.internal_profile_id"
+
+    fun createMarkTopicsCompletedTestIntent(context: Context, internalProfileId: Int): Intent {
```

Comment:
Please add a brief KDoc since this is a public API method.

Reply:
Done.